src/lua/circllhist.lua

```diff
@@ -0,0 +1,565 @@
+--
+-- Copyright (c) 2012-2018, Circonus, Inc. All rights reserved.
+--
+-- Redistribution and use in source and binary forms, with or without
+-- modification, are permitted provided that the following conditions are
+-- met:
+--
+--     * Redistributions of source code must retain the above copyright
+--       notice, this list of conditions and the following disclaimer.
+--     * Redistributions in binary form must reproduce the above
+--       copyright notice, this list of conditions and the following
+--       disclaimer in the documentation and/or other materials provided
+--       with the distribution.
+--     * Neither the name Circonus, Inc. nor the names of its contributors
+--       may be used to endorse or promote products derived from this
+--       software without specific prior written permission.
+--
+-- THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+-- "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+-- LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+-- A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+-- OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+-- SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+-- LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+-- DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+-- THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+-- (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+-- OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+--
+--- Circonus Histogram Module
+--
+-- This module provides access to the Circonus Histogram datatype
+--
+-- - Histogram_C       (relying on libcircllhist via FFI)
+--
+
+local ffi = require('ffi');
+local libhist = require "ffi_libcircllhist"
+
+local Class = function(mt)
+  mt.__index = mt
+  mt.new = function() return setmetatable({}, mt) end
+  return mt
+end
+local string_builder = Class {
+  append = function(self, s) self[#self+1] = tostring(s) end,
+  appendf = function(self, ...) self[#self+1] = string.format(...) end,
+  build = function(self, sep) return table.concat(self, sep or "") end,
+  sort = function(self) table.sort(self) end,
+}
+
+ffi.cdef[[
+void* malloc (size_t size);
+void free (void* ptr);
+]]
+
+-- pl/tablex.lua, MIT/X11
+local tablex = {}
+function tablex.readonly(t)
+  local mt = {
+    __index=t,
+    __newindex=function(t, k, v) error("Attempt to modify read-only table", 2) end,
+    __pairs=function() return pairs(t) end,
+    __ipairs=function() return ipairs(t) end,
+    __len=function() return #t end,
+    __metatable=false
+  }
+  return setmetatable({}, mt)
+end
+
+-- fun.lua, MIT/X11
+local fun = {}
```

Comment:
fun.lua is only used to provide convenient iteration methods. No _need_ to have this here. We provide other means for iteration (cf. below)